gcp_deepvariant_runner.py

```diff
@@ -273,6 +282,56 @@ def _write_actions_to_temp_file(actions):
   return temp_file.name
 
 
+def _get_project_number(project_id):
+  """Returns GCP project number (-1 on failure)."""
+  if hasattr(_get_project_number,
+             'project_number') and _get_project_number.project_number != -1:
+    return _get_project_number.project_number
+  try:
+    args = [
+        'gcloud', 'projects', 'describe', project_id,
+        '--format=value(projectNumber)'
+    ]
+    _get_project_number.project_number = process_util.run_command(
+        args, retries=2)
+  except RuntimeError:
+    # Error is already logged.
+    _get_project_number.project_number = -1
+  return _get_project_number.project_number
+
+
+def report_runtime_metrics(method_name):
+  """Decorator that reports runtime metrics."""
+
+  def decorated(func):
+    """Pseudo decorator."""
+    @functools.wraps(func)
+    def wrapper(pipeline_args, *args, **kwargs):
+      """Wrapper that measures time elapsed, and reports it.
+
+      if stop_collecting_anonymous_usage_metrics is set it only calls the method
+      without collecting any metrics.
+      """
+      if pipeline_args.stop_collecting_anonymous_usage_metrics:
+        func(pipeline_args, *args, **kwargs)
+      else:
+        success = False
```

Comment:
super nit: How about `status = '_Failure',` and after the func succeeds, having status = '_Success', so you can avoid `if else` below.